### DISCUSSION TITLE
Questions on translations

### DISCUSSION THREAD

**Discussion by jimbobbennett** [microsoft/IoT-For-Beginners#290] [Q&A]
I just received an email with some questions on translations, so I thought it best to answer them here in case others also have the same questions:

* Do we create a new branch? How shall we call it ?
* How do I add my collegue as a contributor?
* What is the pull request workflow?
* Is there any other elements we have to have in mind?
* Is there a quality insurance workflow regarding the quality of the translation?

My answers/thoughts:

**Do we create a new branch? How shall we call it**

No need for a new branch. Fork the repo and add your translations to the `main` branch in your fork. You can then raise pull requests to merge your translations in whenever you are ready.

**How do I add my collegue as a contributor**

Contributors show up based off commits, so the best way is for you both to fork and work on different parts of the translation. That way commits from both of you will be included in the contributor list.

**What is the pull request workflow?**

You can raise a PR with as little as a single file translated (or tweaks to an existing file), or multiple files. Lots of smaller PRs are usually easier to process. The translation will be reviewed then merged. The review will ideally be done by someone else who speaks the language, and once they approve someone from Microsoft will add the final approval and merge the PR.

**Is there any other elements we have to have in mind?**

There's some guidance in the [translations.md](https://github.com/microsoft/IoT-For-Beginners/blob/main/TRANSLATIONS.md) file on adding a shield for your translation on the first readme, quizzes and naming conventions.

Make sure all links work - images will be one folder further up for example as the translations are in a subfolder, so make sure the images show. When you have links to other pages (for example in the readme for each project), if the pages are not translated, add links to the English versions, then update to point to the translated one once done.

Use the translation GitHub issues to discuss standards, for example if there is a particular preference for how to translate certain words or phrases. With product names, check the relevant language version of the product page to see how the product name is translated.

@Iftu119 is our translations expert, so can help answer any any questions.

**Is there a quality insurance workflow regarding the quality of the translation?**

This depends on the language. If we only have one translator, then the QA workflow is me and online translation tools. If we have multiple, then. the best way is for one person to translate, and the other review.

**Comment by IterateMe** [comment]
Excellent!
